src/main/java/net/snowflake/ingest/connection/TelemetryService.java

```diff
@@ -66,6 +69,8 @@ public String toString() {
   TelemetryService(CloseableHttpClient httpClient, String clientName, String url) {
     this.clientName = clientName;
     this.telemetry = (TelemetryClient) TelemetryClient.createSessionlessTelemetry(httpClient, url);
+    // At most once every second
+    this.rateLimiter = RateLimiter.create(1.0);
```

Comment:
isnt this too low? essentially it is one per second across all channels IIUC?

Reply:
Good point, what about we create one RateLimiter per channel?

Reply:
yeah.. one per channel is okay!